app/lib/methods/subscriptions/rooms.js

```diff
@@ -121,6 +122,11 @@ const createOrUpdateSubscription = async(subscription, room) => {
 				try {
 					const update = sub.prepareUpdate((s) => {
 						Object.assign(s, tmp);
+						if (subscription.announcement) {
+							if (subscription.announcement !== sub.announcement) {
+								s.bannerClosed = false;
+							}
+						}
```

Comment:
I think you need some logic like this on `findSubscriptionsRooms`, try the follow steps and you'll able to see an error:
* Close the banner
* Close the app
* Change the announcement on other client (Web/Desktop)
* Open the app again
* Banner isn't showed